### DISCUSSION TITLE
Magic home led strip ZJ-WFMN-A RGBW template help

### DISCUSSION THREAD

**Discussion by Rush56** [arendst/Tasmota#17455] [Q&A]
Hi. I flashed my led strip controller for magic home, but i only have lc01, lc06 and lc11 templates avaliable and magic home. None works properly, mixed colors and no white option. Should i compile other tasmota firmware choosing ws2812 leds? Or is there any other way to solve it?

Ive seen some posts from people being able to solve it, followed their solution, but still couldnt get it to work properly. 

It has 8285 chip. 

Thanks in advance.

**Comment by sfromis** [comment]
Those "magic" controllers are a very varied family without every variant "known". What you can do is to experiment to find where the PWM channels are, you can use `SetOption68` to have PWM1-5 separately control (with toggle and slider) each of the channels for red, green, blue, cold white and warm white, respectively, until you've find the right gpios, how many, and right order.

I'd not expect WS2812 config to apply to those, that would be a very different device and situation.

**Reply by Rush56** [reply]
Thank you, will look into it, if i cant find better solution.

Based on this:
https://tasmota.github.io/docs/devices/MagicHome-with-ESP8285/#magichome-led-controllers-aka-flux-led-aka-arilux-al-lc01

I would say lc04 would be the one but no such template in tasmota.

**Reply by sfromis** [reply]
Well, if it is LC04 pin layout, you can try configuring gpio pins as listed on that page. You need no template to do that, that's mainly to make it easier. It is pretty easy to do anyway.

**Reply by Rush56** [reply]
I already tryed, but it did not work. What should i choose as module, not in template setting? If i recall it right, setting it as generic, nothing shows up setting templates.

Im at work now, will take time later and write my findings later :)

Thanks.

**Reply by sfromis** [reply]
Go to Configure Module, choose Generic (18), and then go on to configure each gpio, no templates available as you already observed. When you have a working config, you can convert it to a template, and submit it to the templates repository.

**Reply by Rush56** [reply]
Thanks, will try and report back :)

**Reply by Rush56** [reply]
Well this works, adding it manually as you said for LC04 settings. Thank you!

**Reply by Rush56** [reply]
Id like to ask something else. My fav color is candle like, orange yellow. Now as i set it on higher brightness its nice, but as i lower brightness there is green behind. Any way to change that? 
![IMG_20221220_153042](https://user-images.githubusercontent.com/35640790/208692789-be13017a-ffbb-4614-949f-9d5209b6a75d.jpg)
![IMG_20221220_153025](https://user-images.githubusercontent.com/35640790/208692802-0f39128b-2a50-499c-8b37-ac7bfdb9d369.jpg)

**Reply by sfromis** [reply]
You have a number of features to tweak how the color channels are handled, including the balancing.
https://tasmota.github.io/docs/Lights/#light-options

**Reply by Rush56** [reply]
Great, will look into it. Thank you again, appreciate it, really. Be well, all the best :)
